angular/projects/spark-angular/src/lib/components/sprk-icon/sprk-icon.component.ts

```diff
@@ -3,21 +3,46 @@ import { Component, Input } from '@angular/core';
 @Component({
   selector: 'sprk-icon',
   template: `
-    <svg [ngClass]="getClasses()" viewBox="0 0 64 64" [attr.data-id]="idString">
+    <svg
+      [ngClass]="getClasses()"
+      [attr.viewBox]="getViewBox()"
+      [attr.aria-labelledby]="ariaLabelledby"
+      [attr.data-id]="idString"
+    >
       <use [attr.xlink:href]="icon" />
     </svg>
   `,
-  styles: [':host { line-height: 1; }']
+  styles: [':host { line-height: 1; }'],
 })
 export class SprkIconComponent {
+  /** The viewbox of the svg containing the symbol. */
+  @Input()
+  viewBox: string;
+  /**
+   * Determines which icon is rendered.
+   * Expects the value to match the exact name
+   * of the icon found in the docs
+   * (i.e. `chevron-down`, instead of `chevron down`).
+   */
+  @Input()
+  iconName: string;
+  // TODO: Deprecate on next release
   /**
+   * Deprecated and to be removed next release.
+   * Use `iconName` instead.
    * Determines which icon is rendered.
    * Expects the value to match the exact name
    * of the icon found in the docs
    * (i.e. `chevron-down`, instead of `chevron down`).
    */
   @Input()
   iconType: string;
+  /**
+   * Expects a value to assign to
+   * the `aria-labelledby` attribute of the link.
```

Comment:
Should this be 'icon' instead of 'link'?

Reply:
✅ fixed this and pushed up